### PR TITLE
fix: panic when removing gateway

### PR DESCRIPTION
**Pull request by Xunzhuo**
Fixes: https://github.com/envoyproxy/gateway/issues/619

Signed-off-by: bitliu <bitliu@tencent.com>

### CODE REVIEW

File: internal/provider/kubernetes/gateway.go [hunk 1]
```diff
@@ -377,7 +377,12 @@ func (r *gatewayReconciler) Reconcile(ctx context.Context, request reconcile.Req
 	if !found {
 		gw, ok := r.resources.Gateways.Load(request.NamespacedName)
 		if !ok {
-			r.log.Info("failed to find gateway in the watchable map", "namespace", gw.Namespace, "name", gw.Name)
```
Comment (arkodg) [inline internal/provider/kubernetes/gateway.go:380]: uggh 🙈 , can you just print `request.NamespacedName.String` ?

File: internal/provider/kubernetes/gateway.go [hunk 2]
```diff
@@ -377,7 +377,12 @@ func (r *gatewayReconciler) Reconcile(ctx context.Context, request reconcile.Req
...
+			if gw != nil {
```
Comment (arkodg) [inline internal/provider/kubernetes/gateway.go:380]: not needed, since its already `!ok`

Comment (Xunzhuo) [reply internal/provider/kubernetes/gateway.go:380]: So I think we should return directly here?

File: internal/provider/kubernetes/gateway.go (hunk 1 above)
Comment (Xunzhuo) [reply internal/provider/kubernetes/gateway.go:380]: Yep

File: internal/provider/kubernetes/gateway.go (hunk 2 above)
Comment (Xunzhuo) [reply internal/provider/kubernetes/gateway.go]: When we can load gw from watchable map, we should not let logic to go down, which will cause more panics, I think.

**Review by arkodg — APPROVED** [review]
thanks for the fix !

**Pull request merged**